src/Interfaces/IpTNLPAdapter.cpp

```diff
@@ -1085,7 +1085,16 @@ bool TNLPAdapter::GetSpaces(
       // Get the non zero structure
       Index* g_iRow = new Index[nz_full_jac_g_];
       Index* g_jCol = new Index[nz_full_jac_g_];
-      tnlp_->eval_jac_g(n_full_x_, NULL, false, n_full_g_, nz_full_jac_g_, g_iRow, g_jCol, NULL);
+
+      bool retval = tnlp_->eval_jac_g(n_full_x_, NULL, false, n_full_g_, nz_full_jac_g_, g_iRow, g_jCol, NULL);
+
+      if( !retval )
+      {
+        delete[] g_iRow;
+        delete[] g_jCol;
+
+        return false;
```

Comment:
```suggestion
        ASSERT_EXCEPTION(INVALID_TNLP, "eval_jac_g returned false in GetSpaces");
```
For other failed calls to the TNLP in this function, an exception is thrown, so I would suggest doing the same here.